Makefile

```diff
@@ -143,6 +143,13 @@ build: check-git deps $(PROMU)
 	@echo ">> building Thanos binary in $(PREFIX)"
 	@$(PROMU) build --prefix $(PREFIX)
 
+.PHONY: build-e2e
+build-e2e: ## Build Thanos binary for e2e tests using `promu` with cgo enabled
+build-e2e: check-git deps $(PROMU)
+	@echo ">> building Thanos binary in $(PREFIX)"
+	@$(PROMU) --config=".promu.e2e.yml" build --prefix $(PREFIX)
```

Comment:
Why do we need promu at all? No matter what platform i am running the tests on, the way the tests are configured, only a single image will be used, and that image will always have the same architecture as the machine on which `make test-e2e` was invoked.

It seems to me that promu is overkill/overengineered here. Instead, the Dockerfile.e2e-tests should directly embed the `go build` command with all of the necessary flags. cc @matej-g

Reply:
Actually I was following the convention, since the normal build command in the makefile is using promu and just to maintain homogeneity I used the promu utility here as well, maybe we can have a discussion around this in office hours, whatever is the conclusion, I can make the required changes.

Reply:
My thinking was the same, regardless of whether we need multiplatform build or not - since we're using Promu in the normal build flow, I didn't find it to be too much of an overhead, while being more uniform with the 'main' build flow. If the feeling is this is overengineerd, I don't have a problem with switching to simple `go build`.

Reply:
I guess it's postponed and maybe we'll have it on feb 3, I'll discuss this for sure and get this unblocked

Reply:
I'm not sure if we'll be able to get a clear guidance, maybe we can try to continue with the solution that seem to be agreeable for each of us, i.e. go with the `go build` suggestion? That would help us to unblock ASAP and we can always iterate later on an alternative.

Either way it looks like unblocking the failing E2E tests would need to be addressed before that.

Reply:
@matej-g Okay I'll integrate everything in the docker file itself

Reply:
Hey @matej-g you can check now, all the e2e tests passed locally for me